### PR TITLE
Move hillslope variables off surface dataset

### PR DESCRIPTION
**Pull request by samsrabin**
### Description of changes

This PR moves the variables required for hillslope runs off the surface dataset and onto the new `hillslope_file`. This makes it so we don't have to maintain a hillslope-specific surface dataset for testing purposes. If/when running with hillslopes becomes the default, we can move these variables back onto the surface dataset.

### Specific notes

**Contributors other than yourself, if any:** None

**CTSM Issues Fixed (include github issue #):** None

**Are answers expected to change (and if so in what way)?** No

**Any User Interface Changes (namelist or namelist defaults changes)?** No

**Documentation?** User's Guide documentation will come in on my PR that adds the hillslope scripts to `python/ctsm/` (#2425).

**Testing performed, if any:**
- [x] Initial `hillslope` suite testing.

Waiting until this is next in queue
- [x] Merge in latest master tag
- [x] Derecho and Izumi aux_clm tests are b4b-identical with latest master tag, aside from hillslope tests, which should diff because of new mesh files.

### CODE REVIEW

**Comment by samsrabin** [comment]
Blocked: Waiting on 5.2.

**Comment by samsrabin** [comment]
5.2 is in, so this is no longer blocked.

**Comment by samsrabin** [comment]
I have to change the `5x5_amazon` hillslope tests from `I1850` to `I2000` because there's no default 2000 surface dataset for that resolution. I've thus retargeted this branch at master instead of b4b-dev.

**Comment by samsrabin** [comment]
@swensosc Would you mind having a look over this? Happy to walk you through it if you prefer.

**Comment by swensosc** [comment]
I think most of it looks straightforward (changing input file).  You may wish to add a phrase to this warning "This means all hillslope columns in a gridcell will read identical values" to indicate that this will occur even if the finidat file has hillslope data on it.  Also, is there a check to ensure that hillslope_file is of the same spatial resolution as fsurdat?  If not, is that possible?

**Comment by samsrabin** [comment]
Thanks, @swensosc!
- [ ] Add check that `fsurdat` and `hillslope_file` are the same resolution.
- [ ] Add text to warning message.

File: bld/CLMBuildNamelist.pm [hunk 1]
```diff
@@ -2743,6 +2743,13 @@ SIMYR:    foreach my $sim_yr ( @sim_years ) {
         $log->fatal_error("$useinitvar is being set for you but a $var was not found, so $useinitvar, init_interp_attributes, and finidat must not be set correctly for this configuration in the namelist_default file" );
      }
   }
+
+  # this check has to be here and not earlier since use_init_interp is set here and hillslope is already set above in setup_logic_hillslope
+  if ( &value_is_true($nl->get_value($useinitvar)) && value_is_true($nl->get_value("use_hillslope")) ) {
+     $log->warning("WARNING: You have set use_hillslope while $useinitvar is TRUE.\n This means all hillslope columns in a gridcell will read identical values" .
+                   " from initial conditions. If you are sure you want this behaviour:")
```
Comment (slevis-lmwg) [inline bld/CLMBuildNamelist.pm:2750]: The last sentence seems incomplete. Does the code give instructions for bypassing warnings after the ":"?

**Review by slevis-lmwg — APPROVED** [review]
Looks good to me, @samsrabin

File: bld/CLMBuildNamelist.pm (hunk 1 above)
Comment (samsrabin) [reply bld/CLMBuildNamelist.pm]: Good catch. Resolved with 4b032be5b.

File: bld/CLMBuildNamelist.pm [hunk 2]
```diff
@@ -2759,8 +2759,7 @@ SIMYR:    foreach my $sim_yr ( @sim_years ) {
 
   # this check has to be here and not earlier since use_init_interp is set here and hillslope is already set above in setup_logic_hillslope
   if ( &value_is_true($nl->get_value($useinitvar)) && value_is_true($nl->get_value("use_hillslope")) ) {
-     $log->warning("WARNING: You have set use_hillslope while $useinitvar is TRUE.\n This means all hillslope columns in a gridcell will read identical values" .
-                   " from initial conditions. If you are sure you want this behaviour:")
+     $log->warning("WARNING: You have set use_hillslope while $useinitvar is TRUE.\n This means all hillslope columns in a gridcell will read identical values from initial conditions, even if the initial conditions (finidat) file has hillslope information. If you are sure you want this behaviour, add -ignore_warnings to CLM_BLDNML_OPTS.")
```
Comment (slevis-lmwg) [inline bld/CLMBuildNamelist.pm:2762]: nit: I'm pretty sure that `$log->warning` will generate a default "-ignore_warnings" comment, though I forget whether it tells users where to add the -ignore_warnings...

File: src/main/surfrdMod.F90 [hunk 3]
```diff
@@ -46,6 +46,83 @@ module surfrdMod
 
 contains
 
+  subroutine check_domain_attributes(ncid, begg, endg, ldomain, info)
```
Comment (slevis-lmwg) [inline src/main/surfrdMod.F90:49]: Cool!

**Review by slevis-lmwg — APPROVED** [review]
Thanks, @samsrabin!

**Pull request merged**